src/main/java/org/terasology/blockNetwork/NetworkNode.java

```diff
@@ -19,10 +19,29 @@
 import org.terasology.math.SideBitFlag;
 import org.terasology.math.geom.Vector3i;
 
+/**
+ * Represents a single node in a block network.
+ * Any instance of this class or its children can be used as a node in {@link EfficientNetwork}
+ */
 public class NetworkNode {
+    /**
+     * The location of the Node
+     */
     public final ImmutableBlockLocation location;
+    
+    /**
+     * Based on a combination of the input and output sides, i.e. any side that is an input, output, or both.
+     */
```

Comment:
Again, this will be a comment.

Also, `Based on a combination of the input and output sides`.. what? This sentence feels incomplete. `Based on ... this happens`, or `A variable whose value is based on ...` or something else that makes sense. Remember, this is a comment, whose job is to explain the variable to someone who wants to understand the code.

How does `// Represents the input and output sides of the node, ie, ...` or something along those lines?